File: web/src/engine/playbackPosition.test.ts
// Copyright (C) 2026 Dehalion
// SPDX-License-Identifier: GPL-3.0-only
import { describe, expect, it } from 'vitest'
import { formatCurrent, formatTotal } from './playbackPosition'

describe('playbackPosition', () => {
  it('formats sub-minute duration', () => {
    expect(formatCurrent({ currentMillis: 5_000, totalMillis: 60_000 })).toBe('0:05')
  })

  it('formats minutes and seconds', () => {
    expect(formatCurrent({ currentMillis: 65_000, totalMillis: 200_000 })).toBe('1:05')
  })

  it('pads seconds below ten', () => {
    expect(formatCurrent({ currentMillis: 123_000, totalMillis: 200_000 })).toBe('2:03')
  })

  it('formats total alongside current', () => {
    expect(formatTotal({ currentMillis: 0, totalMillis: 201_000 })).toBe('3:21')
  })

  it('negative total (unknown duration) formats as zero', () => {
    expect(formatTotal({ currentMillis: 0, totalMillis: -1 })).toBe('0:00')
  })
})

File: app/src/test/kotlin/com/stopptanz/app/session/PlaybackPositionTest.kt
// Copyright (C) 2026 Dehalion
// SPDX-License-Identifier: GPL-3.0-only
package com.stopptanz.app.session

import kotlin.test.Test
import kotlin.test.assertEquals

class PlaybackPositionTest {

    @Test
    fun `formats sub-minute duration`() {
        assertEquals("0:05", PlaybackPosition(5_000, 60_000).formatCurrent())
    }

    @Test
    fun `formats minutes and seconds`() {
        assertEquals("1:05", PlaybackPosition(65_000, 200_000).formatCurrent())
    }

    @Test
    fun `pads seconds below ten`() {
        assertEquals("2:03", PlaybackPosition(123_000, 200_000).formatCurrent())
    }

    @Test
    fun `formats total alongside current`() {
        assertEquals("3:21", PlaybackPosition(0, 201_000).formatTotal())
    }

    @Test
    fun `negative total (unknown duration) formats as zero`() {
        assertEquals("0:00", PlaybackPosition(0, -1).formatTotal())
    }
}
